nengo/utils/sockets.py

```diff
@@ -0,0 +1,322 @@
+            self.recv_socket.bind((self.local_addr, self.local_port))
+            self.recv_socket.settimeout(self.timeout)
+        except socket.error:
+            raise RuntimeError("UDPSocket: Could not bind to socket. "
+                               "Address: %s, Port: %s, is in use. If "
+                               "simulation has been run before, wait for "
+                               "previous UDPSocket to release the port. "
+                               "(See 'max_idle_time' argument in class "
+                               "constructor, currently set to %f seconds)" %
+                               (self.local_addr, self.local_port,
+                                self.max_idle_time_current))
+
+    def _retry_connection(self):
+        self._close_recv_socket()
+        while (self.recv_socket is None):
+            time.sleep(self.retry_backoff_time)
+            try:
+                self._open_recv_socket()
+            except socket.error:
+                pass
+            # Failed to open receiving socket, double backoff time, then retry
+            self.retry_backoff_time *= 2
+
+    def _terminate_alive_check_thread(self):
+        self.max_idle_time_current = self.max_idle_time_timeout
+        if self.alive_check_thread is not None:
+            self.last_active = 0
+            self.alive_check_thread.join()
+        self.alive_check_thread = None
+
+    def _close_send_socket(self):
+        if self.send_socket is not None:
+            self.send_socket.close()
+        self.send_socket = None
+
+    def _close_recv_socket(self):
+        if self.recv_socket is not None:
+            self.recv_socket.close()
+        self.recv_socket = None
+
+    def _config_wipe(self):
+        self.local_addr = '127.0.0.1'
+        self.local_port = -1
+        self.dest_addr = '127.0.0.1'
+        self.dest_port = -1
+        self.is_sender = False
+        self.is_receiver = False
+        self.ignore_timestamp = False
+
+        self.close()
+        self._initialize()
+
+    def config_send_only(self, dest_addr, dest_port):
+        self._config_wipe()
+        if (dest_port > 0):
+            self.is_sender = True
+            self.dest_addr = dest_addr
+            self.dest_port = dest_port
+        else:
+            raise ValueError("UDPSocket: Invalid send only configuration."
+                             "Destination port should be > 0")
+
+    def config_recv_only(self, local_port, timeout=5, ignore_timestamp=False):
+        self._config_wipe()
+        if (local_port > 0):
+            self.is_receiver = True
+            self.local_port = local_port
+            self.timeout = timeout
+            self.ignore_timestamp = ignore_timestamp
+        else:
+            raise ValueError("UDPSocket: Invalid recv only configuration."
+                             "Local port should be > 0")
+
+    def config_send_recv(self, local_port, dest_addr, dest_port, timeout=5,
```

Comment:
These functions are never used or tested. What's their intended use?

Reply:
a very good question! took them out and everything still passes, seems fair to assume it's legacy code from some previous iteration...